dali/kernels/slice/slice_cpu.h

```diff
@@ -287,10 +377,32 @@ class SliceCPU {
     return req;
   }
 
-  void Run(KernelContext &context,
-           OutTensorCPU<OutputType, Dims> &out,
-           const InTensorCPU<InputType, Dims> &in,
-           const SliceArgs<OutputType, Dims> &slice_args) {
+  /**
+   * @brief Schedules the kernel work with an execution engine.
+   *
+   *        The work is only schedule and does not start until the user calls RunAll()
+   *        on the execution engine.
+   *
+   *        The user is responsible to synchronize with the execution engine (e.g. WaitForWork())
+   *
+   *        For execution engines other than SequentialExecutionEngine, the algorithm will try
```

Comment:
```suggestion
   *        For execution engines with NumThreads() > 1, the algorithm will try
```

Reply:
this is not true, only SequentialExecutionEngine is specialized